packages/client/lib/commands/PUBSUB_SHARDNUMSUB.ts

```diff
@@ -0,0 +1,24 @@
+import { pushVerdictArguments } from './generic-transformers';
+import { RedisCommandArgument, RedisCommandArguments } from '.';
+
+export const IS_READ_ONLY = true;
+
+export function transformArguments(
+    channels?: Array<RedisCommandArgument> | RedisCommandArgument
+): RedisCommandArguments {
+    const args = ['PUBSUB', 'SHARDNUMSUB'];
+
+    if (channels) return pushVerdictArguments(args, channels);
+
+    return args;
+}
+
+export function transformReply(rawReply: Array<string | number>): Record<string, number> {
+    const transformedReply = Object.create(null);
+
+    for (let i = 0; i < rawReply.length; i +=2) {
```

Comment:
You can use `entries()` to iterate over the array of records.. something like:
```js
 for (const [channel, numSubscribers] of rawReply.entries()) {
    transformedReply[channel] = numSubscribers;
  }
  ```

Reply:
Every first factor of the iterator returned by the entries() is index, so I think it is not an appropriate use 🤔

Reply:
How does "rawReply" look like ?

Reply:
https://redis.io/commands/pubsub-shardnumsub/

Same as reply in pubsub numsub. Repeated channel(string) and count(number)